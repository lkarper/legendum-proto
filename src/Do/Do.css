.Do__container {
    position: relative;
    display: flex;
    flex-direction: column;
    align-items: center;
    height: 100%;
    width: 100%;
}

@media screen and (min-width: 1000px) {
    .Do__container {
        width: 800px;
    }
}