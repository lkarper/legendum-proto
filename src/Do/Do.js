import React, { useState, useEffect } from 'react';
import DoPage from '../DoPage/DoPage';
import ExercisesService from '../services/exercises-service';

const Do = (props) => {
    const { chapt } = props.match.params;

    const [pages, setPages] = useState([]);
    const [page, setPage] = useState(1);
    const [savedUserInput, setSavedUserInput] = useState({});

    useEffect(() => {
        ExercisesService.getExercisesDoByChapter(chapt)
            .then(data => {
                data.sort((a, b) => a.page - b.page);
                setPages(data);
            })
            .catch(error => {
                console.log('error', error);
            });
    }, [chapt]);

    return (
        <div>
            {pages.length ? <h2>{pages[0].exercise_title} {pages[0].exercise_translation}</h2> : <p>Loading...</p>}
            {pages.length ?
                <DoPage
                    data={{
                        chapt,
                        savedUserInput,
                        pages,
                        page,
                        setPage,
                        setSavedUserInput,
                    }}
                />
                : ''
            }
        </div>
    );
}

export default Do;