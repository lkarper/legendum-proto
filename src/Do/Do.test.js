import React from 'react';
import ReactDOM from 'react-dom';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import Do from './Do';

describe('Do component', () => {
    it('renders the UI without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Do />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
        const wrapper = mount(<Do />);
        expect(toJSON(wrapper)).toMatchSnapshot();
        wrapper.unmount();
    });
});
