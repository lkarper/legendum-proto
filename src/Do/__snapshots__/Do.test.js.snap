// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Do component renders the UI as expected 1`] = `
<Do
  match={
    Object {
      "params": Object {
        "chapt": "",
      },
    }
  }
>
  <div
    className="Do__container"
  >
    <LoadingSpinner>
      <div
        className="LoadingSpinner__loading-container"
      >
        <FontAwesomeIcon
          aria-label="Loading."
          className="LoadingSpinner__spinner"
          icon={
            Array [
              "fa",
              "spinner",
            ]
          }
          spin={true}
        >
          <i
            className="fa"
          />
        </FontAwesomeIcon>
      </div>
    </LoadingSpinner>
    <div
      className="Do__alert-div"
      role="alert"
    />
  </div>
</Do>
`;
